cranelift-codegen/src/regalloc/affinity.rs

```diff
@@ -92,10 +92,9 @@ impl Affinity {
                 // to change anything.
                 if constraint.kind != ConstraintKind::Stack && !constraint.regclass.has_subclass(rc)
                 {
-                    // If the register classes don't overlap, `intersect` returns `Unassigned`, and
-                    // we just keep our previous affinity.
+                    // If the register classes don't overlap, we just keep our previous affinity;
```

Comment:
Wording nit: how's `If the register classes overlap, try to shrink our preferred register class.` sound? "try" because there's no guarantee the chosen register class is actually smaller - it could be the same one.